sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java

```diff
@@ -58,76 +60,255 @@ public static TraceConfig getDefault() {
+  public int getMaxNumberOfAttributesPerEvent() {
+    return maxNumberOfAttributesPerEvent;
+  }
 
   /**
    * Returns the global default max number of attributes per {@link Link}.
    *
    * @return the global default max number of attributes per {@link Link}.
    */
-  public abstract int getMaxNumberOfAttributesPerLink();
+  public int getMaxNumberOfAttributesPerLink() {
+    return maxNumberOfAttributesPerLink;
+  }
 
   /**
    * Returns a new {@link Builder}.
    *
    * @return a new {@link Builder}.
    */
-  private static Builder newBuilder() {
-    return new AutoValue_TraceConfig.Builder();
+  public static Builder newBuilder() {
+    return new Builder();
   }
 
   /**
    * Returns a {@link Builder} initialized to the same property values as the current instance.
    *
    * @return a {@link Builder} initialized to the same property values as the current instance.
    */
-  public abstract Builder toBuilder();
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  public static final class Builder extends ConfigBuilder<Builder> {
+    private static final String KEY_SAMPLER_PROBABILITY = "otel.config.sampler.probability";
+    private static final String KEY_SPAN_MAX_NUM_ATTRIBUTES = "otel.config.max.attrs";
+    private static final String KEY_SPAN_MAX_NUM_EVENTS = "otel.config.max.events";
+    private static final String KEY_SPAN_MAX_NUM_LINKS = "otel.config.max.links";
+    private static final String KEY_SPAN_MAX_NUM_ATTRIBUTES_PER_EVENT =
+        "otel.config.max.event.attrs";
+    private static final String KEY_SPAN_MAX_NUM_ATTRIBUTES_PER_LINK = "otel.config.max.link.attrs";
 
-  /** A {@code Builder} class for {@link TraceConfig}. */
-  @AutoValue.Builder
-  public abstract static class Builder {
+    private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES = 32;
+    private static final int DEFAULT_SPAN_MAX_NUM_EVENTS = 128;
+    private static final int DEFAULT_SPAN_MAX_NUM_LINKS = 32;
+    private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_EVENT = 32;
+    private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_LINK = 32;
+
+    private Sampler sampler = Samplers.alwaysOn();
+    private int maxNumberOfAttributes = DEFAULT_SPAN_MAX_NUM_ATTRIBUTES;
+    private int maxNumberOfEvents = DEFAULT_SPAN_MAX_NUM_EVENTS;
+    private int maxNumberOfLinks = DEFAULT_SPAN_MAX_NUM_LINKS;
+    private int maxNumberOfAttributesPerEvent = DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_EVENT;
+    private int maxNumberOfAttributesPerLink = DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_LINK;
+
+    private Builder(TraceConfig traceConfig) {
+      this.sampler = traceConfig.sampler;
+      this.maxNumberOfEvents = traceConfig.maxNumberOfEvents;
+      this.maxNumberOfLinks = traceConfig.maxNumberOfLinks;
+      this.maxNumberOfAttributesPerLink = traceConfig.maxNumberOfAttributesPerLink;
+      this.maxNumberOfAttributesPerEvent = traceConfig.maxNumberOfAttributesPerEvent;
+    }
+
+    private Builder() {}
+
+    /**
```

Comment:
In #1181, we decided to get rid of this documentation from the methods and push it to the class and package-info javadoc